### PR TITLE
Updating user accounts

### PR DESCRIPTION
**Pull request by hedgehog1833**
Updating a user's role and status is now possible via the user administration. I pulled the last checkbox from the card into a new one because it does not really have anything to do with this topic and I wanted to finally get this done here.

### CODE REVIEW

**Comment by hedgehog1833** [comment]
[Update a user account](https://trello.com/c/iEGmTlRI/55-update-a-user-account)

File: src/main/resources/static/js/admin/users.js [hunk 1]
```diff
@@ -176,37 +191,59 @@ function updateUser () {
  * Sends the update request to the server.
  */
 function sendUpdateUserRequest() {
-    // ToDo: https://trello.com/c/iEGmTlRI
+    $.post({
+        url: '/rest/v1/users',
+        data: createUpdateUserRequest(),
+        type: 'POST',
```
Comment (DanielW1987) [inline src/main/resources/static/js/admin/users.js]: PUT

File: src/main/resources/static/js/admin/users.js [hunk 2]
```diff
@@ -117,11 +119,12 @@ function onCreateAdministratorSuccess(createResponse) {
...
 /**
  * Error callback for creating a new administrator.
- * @param errorResponse The json response
+ * @param errorResponse     The json response
+ * @param validationAreaId  ID of the area to display errors (create/update)
  */
-function onCreateAdministratorError(errorResponse) {
-    resetCreateAdminUserValidationArea();
-    const validationMessageArea = $('#create-admin-user-validation-area');
+function onCreateUpdateError(errorResponse, validationAreaId) {
```
Comment (DanielW1987) [inline src/main/resources/static/js/admin/users.js]: I don't like that kind of mix in the name. I'd rather have one method more:

- onCreateError (calls onError)
- onUpdateError (calls onError)
- onError

File: src/test/java/com/metalr2/service/user/UserServiceTest.java [hunk 3]
```diff
@@ -377,11 +407,56 @@ void update_user_for_not_existing_user() {
...
+
+    when(currentUserSupplier.get()).thenReturn(user);
+    when(userRepository.findByPublicId(PUBLIC_ID)).thenReturn(Optional.of(user));
+
+    // when
+    Throwable throwable = catchThrowable(() -> userService.updateUser(PUBLIC_ID, userDtoForUpdate));
+
+    // then
+    verify(userRepository, times(1)).findByPublicId(PUBLIC_ID);
+    verify(currentUserSupplier, times(1)).get();
```
Comment (DanielW1987) [inline src/test/java/com/metalr2/service/user/UserServiceTest.java]: you already have test the method invocation in other tests. The test should be as minimal as possible to test the exception.

File: src/test/java/com/metalr2/service/user/UserServiceTest.java [hunk 4]
```diff
@@ -377,11 +407,56 @@ void update_user_for_not_existing_user() {
...
```
Comment (DanielW1987) [inline src/test/java/com/metalr2/service/user/UserServiceTest.java]: you already have test the method invocation in other tests. The test should be as minimal as possible to test the exception.

**Pull request merged**